code/game/machinery/computer/launchpad_control.dm

```diff
@@ -0,0 +1,155 @@
+/obj/machinery/computer/launchpad
+	name = "\improper Launchpad Control Console"
+	desc = "Used to teleport objects to and from a launchpad."
+	icon_screen = "teleport"
+	icon_keyboard = "teleport_key"
+	circuit = /obj/item/weapon/circuitboard/computer/launchpad_console
+	var/sending = TRUE
+	var/current_pad //current pad viewed on the screen
+	var/list/obj/machinery/launchpad/launchpads
+	var/maximum_pads = 4
+
+/obj/machinery/computer/launchpad/Initialize()
+	launchpads = list()
+	. = ..()
+
+/obj/machinery/computer/launchpad/attack_paw(mob/user)
+	to_chat(user, "<span class='warning'>You are too primitive to use this computer!</span>")
+	return
+
+/obj/machinery/computer/launchpad/attackby(obj/item/W, mob/user, params)
+	if(istype(W, /obj/item/device/multitool))
+		var/obj/item/device/multitool/M = W
+		if(M.buffer && istype(M.buffer, /obj/machinery/launchpad))
+			if(LAZYLEN(launchpads) < maximum_pads)
+				launchpads |= M.buffer
+				M.buffer = null
+				to_chat(user, "<span class='notice'>You upload the data from the [W.name]'s buffer.</span>")
+			else
+				to_chat(user, "<span class='warning'>[src] cannot handle any more connections!</span>")
+	else
+		return ..()
+
+/obj/machinery/computer/launchpad/attack_ai(mob/user)
+	attack_hand(user)
+
+/obj/machinery/computer/launchpad/attack_hand(mob/user)
+	if(..())
+		return
+	interact(user)
+
+/obj/machinery/computer/launchpad/proc/pad_exists(number)
+	var/obj/machinery/launchpad/pad = launchpads[number]
+	if(QDELETED(pad))
+		return FALSE
+	return TRUE
+
+/obj/machinery/computer/launchpad/proc/get_pad(number)
+	var/obj/machinery/launchpad/pad = launchpads[number]
+	return pad
+
+/obj/machinery/computer/launchpad/interact(mob/user)
+	var/t
```

Comment:
Wherever you're combining strings together, use a list (```var/list/t```) and then ```t.Join()``` when you want it as a finalised string.

Do this for the rest of the PR